### DISCUSSION TITLE
Argument type not correct for with-query with closure

### DISCUSSION THREAD

**Discussion by joostdebruijn** [larastan/larastan#2158] [Q&A] [answered]
I've a query with a `with`, like this:

```php
$query->with([
    'groups' => function (HasMany $query): void {
        $query->orderBy('order');
    },
]);
```

This returns the following error since v3 (PHPstan v2.0.4, Larastan 3.0.2, running at level 6):

```
Parameter #1 $relations of method Illuminate\Database\Eloquent\Builder<App\Models\GroupType>::with() expects array<array|(Closure(Illuminate\Database\Eloquent\Relations\Relation<*, *, *>):  
         mixed)|string>|string, array{groups: Closure(Illuminate\Database\Eloquent\Relations\HasMany): void} given.
```

The code itself is working correctly and it's a closure with a `Relation` as parameter, so I'm note sure what I need to change to fix is error. Can someone give me a pointer? :)

**Comment by calebdw** [comment, marked as answer]
Remove the HasMany type from the closure

**Reply by calebdw** [reply]
Parameter types are not covariant

**Comment by joostdebruijn** [comment]
@calebdw Thanks for the very quick response. :)
